subsys/net/conn_mgr/conn_mgr_connectivity.c

```diff
@@ -235,8 +235,52 @@ int conn_mgr_if_set_timeout(struct net_if *iface, int timeout)
 	return 0;
 }
 
+/* Automated behavior handling */
+
+/**
+ * @brief Perform automated behaviors in response to ifaces going admin-up.
+ *
+ * @param iface - The iface which became admin-up.
+ */
+static void conn_mgr_conn_handle_iface_admin_up(struct net_if *iface)
+{
+	int err;
+
+	/* Ignore ifaces that don't have connectivity implementations */
+	if (!conn_mgr_if_is_bound(iface)) {
+		return;
+	}
+
+	/* Ignore ifaces for which auto-connect is disabled */
+	if (conn_mgr_if_get_flag(iface, CONN_MGR_IF_NO_AUTO_CONNECT)) {
+		return;
+	}
+
+	/* Otherwise, automatically instruct the iface to connect */
+	err = conn_mgr_if_connect(iface);
+	if (err < 0) {
+		NET_ERR("iface auto-connect failed: %d", err);
+	}
+}
+
+static struct net_mgmt_event_callback conn_mgr_conn_iface_cb;
+static void conn_mgr_conn_iface_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+					struct net_if *iface)
+{
+	if ((mgmt_event & CONN_MGR_CONN_IFACE_EVENTS_MASK) != mgmt_event) {
+		return;
+	}
+
+	switch (NET_MGMT_GET_COMMAND(mgmt_event)) {
```

Comment:
Getting `NET_MGMT_GET_COMMAND()` involved is not really needed, just check for `NET_EVENT_IF_ADMIN_UP`.